src/renderers/shared/stack/reconciler/ReactCompositeComponent.js

```diff
@@ -488,23 +458,21 @@ var ReactCompositeComponentMixin = {
 
   performInitialMount: function(renderedElement, hostParent, hostContainerInfo, transaction, context) {
     var inst = this._instance;
+
+    var debugID = 0;
+    if (__DEV__) {
+      debugID = this._debugID;
+    }
+
     if (inst.componentWillMount) {
-      if (__DEV__) {
-        if (this._debugID !== 0) {
-          ReactInstrumentation.debugTool.onBeginLifeCycleTimer(
-            this._debugID,
-            'componentWillMount'
-          );
-        }
-      }
-      inst.componentWillMount();
-      if (__DEV__) {
-        if (this._debugID !== 0) {
-          ReactInstrumentation.debugTool.onEndLifeCycleTimer(
-            this._debugID,
-            'componentWillMount'
-          );
-        }
+      if (__DEV__ && debugID !== 0) {
+        measureLifeCyclePerf(
+          () => inst.componentWillMount(),
```

Comment:
I wish that JS had some compile time ways of doing zero-cost abstractions instead of having to copy some non trivial logic twice. In C I would have used a macro for it.